source/reference/about-mongosync.txt

```diff
@@ -0,0 +1,142 @@
+
+You must create an initial connection to the source and destination 
+clusters before ``mongosync`` can start to sync data. For more information, see 
+:ref:`c2c-connecting`.
+
+When ``mongosync`` connects to the source and destination clusters, the 
+synchronization process enters the :ref:`IDLE <c2c-state-idle>` state.  
+
+During Sync 
+~~~~~~~~~~~
+
+To start the synchronization process, call the :ref:`start <c2c-api-start>` 
+endpoint. 
+
+The ``start`` endpoint begins the ``RUNNING`` state, which is when ``mongosync`` 
+syncs data from the source cluster to the destination cluster through the 
+:ref:`collection copy <c2c-collection-copy>` process. If you make subsequent 
+writes to the source cluster, ``mongosync`` applies the operations to the 
+destination cluster through the :ref:`change event application <c2c-cea>` 
+process.
+
+For every collection on the source, ``mongosync`` creates a corresponding 
+collection on the destination. 
+
+``mongosync`` temporarily alters some collection characteristics during 
+the ``RUNNING`` state and changes these back to their source 
+cluster characteristics during the ``COMMIT`` state. For more information, see
+:ref:`Commit Alters <c2c-commit-alters>`.
+
+.. _c2c-collection-copy:
+
+Collection Copy
+'''''''''''''''
+
+The collection copy process is the initial full data transfer that copies each 
+collection from the source cluster to the destination cluster. ``mongosync`` 
+performs collection copy by splitting each source collection into partitions 
+and copying all the partitions to their corresponding destination collections.
+
+.. _c2c-cea:
+
+Change Event Application
+''''''''''''''''''''''''
+
+The change event application (CEA) process continuously updates 
+the destination cluster when you make changes to the source cluster. If there 
+are multiple ``mongosync`` instances, all instances finish 
+collection copy before starting the CEA process.
+
+During CEA, ``mongosync`` receives :ref:`change stream <changeStreams>` events 
+from the source cluster, converts these events into writes, and applies these 
+writes on the destination cluster. 
+
+Pausing Sync
+~~~~~~~~~~~~
+
+To pause the sync process and enter the :ref:`PAUSED <c2c-state-paused>` state,
+call the :ref:`pause <c2c-api-pause>` endpoint. 
+
+To resume a paused synchronization session and return ``mongosync`` to the 
+``RUNNING`` state, call the :ref:`resume <c2c-api-resume>` endpoint. 
+
+Finalizing Sync
+~~~~~~~~~~~~~~~
+
+To finalize the sync between the source and destination clusters, 
+call the :ref:`commit <c2c-api-commit>` endpoint. 
+
+The ``commit`` endpoint starts the :ref:`COMMITTING <c2c-state-committing>` 
+state. ``mongosync`` stops continuous sync between the source and desitnation 
+clusters. It also restores collection characteristics that ``mongosync`` 
+temporarily alters during synchronization. 
+
+Once you call the ``commit`` endpoint, ``mongosync`` halts any writes on the 
```

Comment:
The order should be reversed. Mongosync retrieves the timestamp of the most recent operation in the source cluster and applies the remaining change events from the source up to that timestamp.

Reply:
Which order did you clarified? I feel there's a bit of misunderstanding here. I'm happy to discuss more on slack if you would like.

There're two things you mentioned in this sentence: "halts any writes on the **source** cluster" and "retrieves the timestamp of the most recent operation on the source cluster". The first thing isn't correct because mongosync shouldn't write to the source cluster at any point. Are you referring to [the write blocking step](https://github.com/10gen/mongosync/blob/main/docs/architecture/committing.md?plain=1#L8) in the internal doc? If you meant "halts any writes on the **destination** cluster", that should happen after retrieving the timestamp. 

To answer your question: change events are continuously applied in CEA, so it doesn't matter if a user receives a success response from the `/commit` endpoint or not. However, CEA exits after `/commit` endpoint returns a success message.

Reply:
Resolved over slack!